src/Console/Shell.php

```diff
@@ -307,6 +310,7 @@ public function dispatchShell()
         if (is_string($args[0]) && count($args) === 1) {
             $args = explode(' ', $args[0]);
         }
+        $args[] = '--requested';
```

Comment:
@markstory Took a deeper look at your suggestion. Something like that could work :

- Change dispatchShell() so it can accept an array of parameters allowing those extra parameters to be passed :
```php
Shell::dispatchShell([
    'command' => 'schema create DbAcl'
    'extra' => [
        'requested' => true
    ]
])
```
``command`` would be parsed as the current arguments are parsed and ``extra`` passed to the ``ShellDispatcher`` (which would have its constructor modified to accept them as well as a new property to store them until the ``Shell`` is dispatched)
Current signature could still work.
- Extra parameters would be passed to the Shell either through its constructor or a dedicated method called before dispatch.
- Have a new ``extra`` property on ``Shell`` (or a new dimension in ``param``)
- Make ``Shell::param()`` able to access ``extra`` param via something like ``Shell::param('requested', 'extra')``. 

This way, no new flags are added and it is possible to add others in the future and gives the ability to users to have their own extra parameters for their Shells without having to create a new flag.
This would however allow those extra parameters to be used only with ``dispatchShell()`` and prevent users from using them in a "normal call" (for instance, with the current states of the changes, it is possible to call ``cake my_shell my_command --requested`` and not have the welcome message displayed). But I don't know if it has any real value though so I'm okay with that.

This has not been tested or coded. Just a draft of ideas. What do you think ?